### PR TITLE
[NOT FOR MERGE] Kafka to pub sub e2e run actual pipeline

### PR DESCRIPTION
**Pull request by ramazan-yapparov**
Changed test so it runs actual pipeline instead of building one in test.

Added `KafkaConsumerConfig` option for passing additional Kafka parameters.

Moved `Pipeline` object creation to the `KafkaToPubsub#main` method so we could pass test pipeline instance into the `KafkaToPubsub#run` method in the test.

### CODE REVIEW

File: examples/java/src/main/java/org/apache/beam/examples/complete/kafkatopubsub/KafkaToPubsub.java [hunk 1]
```diff
@@ -151,17 +152,19 @@ public static void main(String[] args) {
...
   /**
    * Runs a pipeline which reads message from Kafka and writes it to GCS.
    *
    * @param options arguments to the pipeline
    */
-  public static PipelineResult run(KafkaToPubsubOptions options) {
+  public static PipelineResult run(Pipeline pipeline, KafkaToPubsubOptions options) {
     // Configure Kafka consumer properties
-    Map<String, Object> kafkaConfig = new HashMap<>();
+    Map<String, Object> kafkaConfig = parseKafkaConsumerConfig(options.getKafkaConsumerConfig());
```
Comment (KhaninArtur) [inline examples/java/src/main/java/org/apache/beam/examples/complete/kafkatopubsub/KafkaToPubsub.java:167]: I think it is better to leave the initialization of `kafkaConfig` with an empty `HashMap` and then update it with the result of `parseKafkaConsumerConfig` function. Such explicit initialization may eliminate the necessity to immediately go to the function and see what's going on in it.

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 2]
```diff
@@ -616,6 +616,7 @@ class BeamModulePlugin implements Plugin<Project> {
         testcontainers_kafka                        : "org.testcontainers:kafka:$testcontainers_version",
         testcontainers_localstack                   : "org.testcontainers:localstack:$testcontainers_version",
         testcontainers_postgresql                   : "org.testcontainers:postgresql:$testcontainers_version",
+        testcontainers_gcloud                       : "org.testcontainers:gcloud:$testcontainers_version",
```
Comment (KhaninArtur) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:619]: Do I understand right that `testcontainers_gcloud` provides us Pub/Sub emulator?

File: examples/java/src/test/java/org/apache/beam/examples/complete/kafkatopubsub/KafkaToPubsubE2ETest.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  }
+
+  @Test
+  public void testKafkaToPubsubE2E() throws IOException, InterruptedException {
+    PipelineResult job = KafkaToPubsub.run(pipeline, OPTIONS.as(KafkaToPubsubOptions.class));
+
+    sendKafkaMessage();
+    testPubsub.assertThatTopicEventuallyReceives(
+        hasProperty("payload", equalTo(PUBSUB_MESSAGE.getBytes(StandardCharsets.UTF_8)))
+    ).waitForUpTo(Duration.standardSeconds(25));
```
Comment (KhaninArtur) [inline examples/java/src/test/java/org/apache/beam/examples/complete/kafkatopubsub/KafkaToPubsubE2ETest.java:94]: 25 seconds seem a bit unnatural. What do you think about replacing it with 30 seconds?

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsubSignal.java [hunk 4]
```diff
@@ -74,7 +74,7 @@
  * <p>Uses a random temporary Pubsub topic for synchronization.
  */
 @SuppressWarnings({
-  "nullness" // TODO(https://issues.apache.org/jira/browse/BEAM-10402)
+    "nullness" // TODO(https://issues.apache.org/jira/browse/BEAM-10402)
```
Comment (KhaninArtur) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsubSignal.java:77]: Probably, we can revert this change.

**Review by KhaninArtur — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 2 above)
Comment (ramazan-yapparov) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:619]: Yes, it does

File: examples/java/src/test/java/org/apache/beam/examples/complete/kafkatopubsub/KafkaToPubsubE2ETest.java (hunk 3 above)
Comment (ramazan-yapparov) [reply examples/java/src/test/java/org/apache/beam/examples/complete/kafkatopubsub/KafkaToPubsubE2ETest.java:94]: I'll increase it to 1 minute

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsubSignal.java (hunk 4 above)
Comment (ramazan-yapparov) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsubSignal.java]: Reverted

File: examples/java/src/main/java/org/apache/beam/examples/complete/kafkatopubsub/KafkaToPubsub.java (hunk 1 above)
Comment (ramazan-yapparov) [reply examples/java/src/main/java/org/apache/beam/examples/complete/kafkatopubsub/KafkaToPubsub.java]: Fixed

**Review by KhaninArtur — APPROVED** [review]

**Review by KhaninArtur — APPROVED** [review]
LGTM. Just a friendly reminder - please, don't merge this PR into our fork.

**Comment by ramazan-yapparov** [comment]
Review is complete, closing PR
